storage.go

```diff
@@ -4,229 +4,156 @@ import (
 	"context"
 	"encoding/json"
 	"strings"
-	"time"
 
 	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/giantswarm/microstorage"
-	"github.com/giantswarm/operatorkit/tpr"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	apismeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	api "k8s.io/client-go/pkg/api/v1"
 )
 
-const (
-	tpoCreateMaxElapsedTime = 30 * time.Second
-)
-
-type CRDConfig struct {
-	Name, Version, Description string
-}
-
-type TPOConfig struct {
-	Name, Namespace string
-}
-
 type Config struct {
-	// Dependencies.
-
+	CRDClient *k8scrdclient.CRDClient
+	G8sClient versioned.Interface
```

Comment:
Should we name it G8sClient? It would be nice to change the example in k8srestconfig (operatorkit) package godoc as well.

Reply:
I name it `G8sClient` here. Not sure what to rename now. Do you mean the `versioned` import path? AFAIK this is generated by the k8s code generator.

Reply:
https://github.com/giantswarm/operatorkit/blob/ac0c48942edfd1acaae8c0fec1c4ce7e2a043cad/client/k8srestconfig/k8s_rest_config.go#L47

It'd be nice to align names. I like g8sClient. Let's use the same name everywhere 🌍

Reply:
I don't like it to alias packages. I think it is backwards. We should rather try to get the package name right, which in this case means we should contribute to the code generator somehow. How important is this for you? I would like to keep it as it is and move on.

Reply:
This is ok. What I mean is to replace this line in the godoc:

```
gsClient, err := gsclient.NewForConfig(restConfig)
```

with

```
g8sClient, err := g8sclient.NewForConfig(restConfig)
```

or (if you like)

```
g8sClient, err := versioned.NewForConfig(restConfig)
```

Reply:
Done in https://github.com/giantswarm/operatorkit/pull/140.

Reply:
Not really important though.